Samples/kotlin/src/main/java/com/phunware/kotlin/sample/utils/extensions/Context.kt

```diff
@@ -0,0 +1,19 @@
+package com.phunware.kotlin.sample.utils.extensions
+
+import android.content.Context
+import android.util.TypedValue
+import com.phunware.kotlin.sample.R
+
+/**
+ * Returns the accent color as a ColorInt for the theme on the current context.
+ */
+fun Context.accentColor(): Int {
+    val typedValue = TypedValue()
+
+    val a = obtainStyledAttributes(typedValue.data, intArrayOf(R.attr.colorAccent))
+    val color = a.getColor(0, 0)
+
+    a.recycle()
```

Comment:
do you mind taking this to a be a teaching moment?
I just learned that TypedValue is "_Container for a dynamically typed data value_"
val a here should be a TypedArray if I read the dac correctly.
Why did you call recycle? is this something that we should be doing for other things?

Reply:
When you grab a styled attributes reference from a context, it must always be "recycled" after you use it.  

Android docs simply state:

> Note that TypedArray objects are a shared resource and must be recycled after use.

My understanding is that it's basically an object that holds the values for every single possible theme attribute. It's expensive to create, so Android uses an object pool for this -- when you are done with something in an object pool you return it back to the pool for someone else to use.

Reply:
Thank you!